larpandoradlcontent/LArSignalId/DlSNSignalAlgorithm.cc

```diff
@@ -0,0 +1,1060 @@
+            PANDORA_MONITORING_API(VisualizeCaloHits(this->GetPandora(),
+                                     &photonCandidatesW, "true photon W", BLACK));
+            PANDORA_MONITORING_API(VisualizeCaloHits(this->GetPandora(),
+                                     &electronCandidatesW, "true electron W", RED));
+         }
+
+         PANDORA_MONITORING_API(SetEveDisplayParameters(this->GetPandora(), true, DETECTOR_VIEW_XZ, -1.f, 1.f, 1.f));
+         PANDORA_MONITORING_API(ViewEvent(this->GetPandora())); 	    
+    }
+    }
+    
+
+    if (signalCandidatesU.empty() || signalCandidatesV.empty() || signalCandidatesW.empty() || signalCandidates2D.empty())
+    {
+        std::cout << "Error: A CaloHitList is empty" << std::endl;
+    }
+    if (!signalCandidatesU.empty())                                                                                        
+        PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::SaveList(*this, signalCandidatesU, m_signalListNameU));
+    
+    if (!signalCandidatesV.empty())
+        PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::SaveList(*this, signalCandidatesV, m_signalListNameV));
+    
+    if (!signalCandidatesW.empty())
+        PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::SaveList(*this, signalCandidatesW, m_signalListNameW));
+    
+    if (!signalCandidates2D.empty())
+        PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::SaveList(*this, signalCandidates2D, m_signalListName2D));
+
+    if (!backgroundCaloHitList.empty())
+        PANDORA_THROW_RESULT_IF(STATUS_CODE_SUCCESS, !=, PandoraContentApi::SaveList(*this, backgroundCaloHitList, m_backgroundListName));
+
+    return STATUS_CODE_SUCCESS;
+}
+
+//-----------------------------------------------------------------------------------------------------------------------------------------
+
+StatusCode DlSNSignalAlgorithm::MakeNetworkInputFromHits(const CaloHitList &caloHits, const HitType view, const float xMin,
+    const float xMax, const float zMin, const float zMax, LArDLHelper::TorchInput &networkInput, PixelMap &pixelMap) const
+{
+    // ATTN If wire w pitches vary between TPCs, exception will be raised in initialisation of lar pseudolayer plugin
+    const LArTPC *const pTPC(this->GetPandora().GetGeometry()->GetLArTPCMap().begin()->second);
+    const float pitch(view == TPC_VIEW_U ? pTPC->GetWirePitchU() : view == TPC_VIEW_V ? pTPC->GetWirePitchV() : pTPC->GetWirePitchW());
+
+    // Determine the bin edges
+    std::vector<double> xBinEdges(m_width + 1);
+    std::vector<double> zBinEdges(m_height + 1);
+    xBinEdges[0] = xMin - 0.5f * m_driftStep;
+    const double dx = ((xMax + 0.5f * m_driftStep) - xBinEdges[0]) / m_width;
+    for (int i = 1; i < m_width + 1; ++i)
+        xBinEdges[i] = xBinEdges[i - 1] + dx;
+    zBinEdges[0] = zMin - 0.5f * pitch;
+    const double dz = ((zMax + 0.5f * pitch) - zBinEdges[0]) / m_height;
```

Comment:
Might be overkill, but is there any defence against user error making width and height 0?